File: exercism/javascript/grade-school/grade-school.js
class School {
  constructor() {
    this._roster = {};
  }

  add(name, grade) {
    this._roster = {
      ...this._roster,
      [grade]: this.grade(grade)
        .concat(name)
        .sort()
    };
  }

  grade(grade) {
    return this.roster()[grade] || [];
  }

  roster() {
    return JSON.parse(JSON.stringify(this._roster));
  }
}

export default School;

File: exercism/clojure/grade-school/src/grade_school.clj
(ns grade-school)

(defn add-student [grade name]
  (vec (conj grade name)))

(defn add [school name grade]
  (update school grade add-student name))

(defn grade [school grade]
  (get school grade []))

(defn sort-grade-students [[grade students]]
  {grade (sort students)})

(defn sorted [school]
  (->> school
    (map sort-grade-students)
    (into (sorted-map))))
